compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/staticcodeanalyzer/HttpServiceDeclarationAnalyzer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler.staticcodeanalyzer;
+
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.scan.Reporter;
+import io.ballerina.stdlib.http.compiler.HttpCompilerPluginUtil;
+
+class HttpServiceDeclarationAnalyzer extends HttpServiceAnalyzer {
```

Comment:
Can't we combine `HttpServiceClassAnalyzer`, `HttpServiceDeclarationAnalyzer` and `HttpServiceObjectTypeAnalyzer` classes? Seems like they have similar code

Reply:
Yes, we can achieve that by registering a class like this:
```java
analysisContext.addSyntaxNodeAnalysisTask(new ServiceAnalyzer(reporter), 
    List.of(CLASS_DEFINITION, OBJECT_TYPE_DESC, SERVICE_DECLARATION));
```
However, using a single ServiceAnalyzer would require multiple if-checks in the perform method to handle different syntax kinds. Instead, I preferred using polymorphism over conditional checks and moved common functionality to an abstract HttpServiceAnalyzer class. WDYT?

Reply:
How about defining an interface to these common types which exposes the common properties? 

I have did something similar for resource function implementation and resource function definition (Interface - `ResourceFunction`, Implementations - `ResourceFunctionDefintion` and `ResourceFunctionDeclaration`)
Can you check whether it is viable here?

Reply:
Refactored [2aa962f](https://github.com/ballerina-platform/module-ballerina-http/pull/2202/commits/2aa962f18f1fcf259c65a9deae84606a1b2b4dad)